docs/styleguide.md

```diff
@@ -0,0 +1,24 @@
+
+List of programming style rules
+===============================
+
+(will be re-structured as it grows)
+
+- qualified imports are given the name of the module (without path),
+  not an abbreviation (`import qualified Data.Map as Map`).
+
```

Comment:
This is going to lead to hugely unhelpful import names in many cases (e.g. `Data.Map.Lazy` and friends).